### ISSUE TITLE
Funding ScriptSignatureParams did not match offer funding inputs

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#3026]
This is trying to settle a DLC that is related to this offer  

version: 0.5.0-176-bb8b3c01-ADAPTOR-ECDSA-DLC-SNAPSHOT

https://oracle.suredbits.com/event/442f0587e56774850b4231c2f235d72601d7cba590191ca8cfc4001b06652694

I think this is because i might not be synced ? See the missing funding txid field.

![Screenshot from 2021-05-04 10-40-23](https://user-images.githubusercontent.com/3514957/117030429-428cf180-acc5-11eb-8c06-60a78f915732.png)

```
[info] 2021-05-04T15:37:12UTC INFO [Server] Unhandled error in server:
[info] java.lang.IllegalArgumentException: requirement failed: Funding ScriptSignatureParams did not match offer funding inputs
[info]  at scala.Predef$.require(Predef.scala:338)
[info]  at org.bitcoins.dlc.sign.DLCTxSigner.<init>(DLCTxSigner.scala:53)
[info]  at org.bitcoins.dlc.wallet.DLCWallet.$anonfun$signerFromDb$5(DLCWallet.scala:1334)
[info]  at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:431)
[info]  at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:63)
[info]  at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:100)
[info]  at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[info]  at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)
[info]  at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:100)
[info]  at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:49)
[info]  at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:48)
[info]  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
[info]  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
[info]  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
[info]  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
[info]  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
[info] 2021-05-04T15:37:12UTC INFO [ActorSystemImpl] http-rpc-server: Response for
```

It seems I am synced
```
bitcoin-s-cli getblockcount 
681908
```

I think we may have missed the funding transaction when i started the server up?

and the tx is confirmed in the chain

**Comment by Christewart** [comment]
I think there may be some sort of syncing done in the background? Idk I get these logs consisently, but they aren't useful. I've added them to the bottom of this issue.

here is my wallet state

```
 bitcoin-s-cli walletinfo
{
  "wallet": {
    "keymanager": {
      "rootXpub": ""
    },
    "walletName": "old-wallet",
    "xpub": "...",
    "hdPath": "m/84'/0'/0'",
    "height": 681908,
    "blockHash": "00000000000000000006eb56b96034c1cfae6b3716a2360541c5ff94b574de65"
  }
}
```

and I am seeing logs about syncing blocks!

>[info] 2021-05-04T15:51:14UTC INFO [DLCWallet$DLCWalletImpl] Finished processing of block=0000000000000000000b1c77241c6a7d1a3da81d25980ad0bb7f7254eb9b47c8. It took 27471ms

So it seems like our wallet state is incorrect?


```
[info]   Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/11.0.11, Host, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,83 bytes total),HttpProtocol(HTTP/1.1))
[info]   Response: Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(application/json,40 bytes total),HttpProtocol(HTTP/1.1)))
[info] 2021-05-04T15:48:07UTC INFO [ActorSystemImpl] http-rpc-server: Response for
[info]   Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/11.0.11, Host, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,83 bytes total),HttpProtocol(HTTP/1.1))
[info]   Response: Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(application/json,40 bytes total),HttpProtocol(HTTP/1.1)))
[info] 2021-05-04T15:48:17UTC INFO [ActorSystemImpl] http-rpc-server: Response for
[info]   Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/11.0.11, Host, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,83 bytes total),HttpProtocol(HTTP/1.1))
[info]   Response: Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(application/json,40 bytes total),HttpProtocol(HTTP/1.1)))
[info] 2021-05-04T15:48:27UTC INFO [ActorSystemImpl] http-rpc-server: Response for
[info]   Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/11.0.11, Host, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,83 bytes total),HttpProtocol(HTTP/1.1))
[info]   Response: Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(application/json,40 bytes total),HttpProtocol(HTTP/1.1)))
[info] 2021-05-04T15:48:37UTC INFO [ActorSystemImpl] http-rpc-server: Response for
[info]   Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/11.0.11, Host, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,83 bytes total),HttpProtocol(HTTP/1.1))
[info]   Response: Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(application/json,40 bytes total),HttpProtocol(HTTP/1.1)))
[info] 2021-05-04T15:48:47UTC INFO [ActorSystemImpl] http-rpc-server: Response for
[info]   Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/11.0.11, Host, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,83 bytes total),HttpProtocol(HTTP/1.1))
[info]   Response: Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(application/json,40 bytes total),HttpProtocol(HTTP/1.1)))
```

**Comment by benthecarman** [comment]
> I think there may be some sort of syncing done in the background? Idk I get these logs consisently, but they aren't useful

This is the bitcoind polling every 10 seconds for new blocks

**Comment by Christewart** [comment]
Rescanning did not work. I do have the funding txid now though! 

![Screenshot from 2021-05-04 12-24-49](https://user-images.githubusercontent.com/3514957/117044170-bf26cc80-acd3-11eb-8e84-2100840b58dc.png)

**Comment by Christewart** [comment]
I tried with #3029 with this result

```
2021-05-04T18:40:27UTC INFO [Server] Unhandled error in server:
java.lang.IllegalArgumentException: requirement failed: Funding ScriptSignatureParams did not match offer funding inputs
        at scala.Predef$.require(Predef.scala:338)
        at org.bitcoins.dlc.sign.DLCTxSigner.<init>(DLCTxSigner.scala:56)
        at org.bitcoins.dlc.wallet.DLCWallet.$anonfun$signerFromDb$5(DLCWallet.scala:1334)
        at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:431)
        at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:63)
        at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:100)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)
        at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:100)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:49)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:48)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
2021-05-04T18:40:27UTC INFO [ActorSystemImpl] http-rpc-server: Response for
  Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/11.0.11, Host, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,422 bytes total),HttpProtocol(HTTP/1.1))
  Response: Complete(HttpResponse(500 Internal Server Error,List(),HttpEntity.Strict(application/json,126 bytes total),HttpProtocol(HTTP/1.1)))
java.lang.RuntimeException: Request failed: requirement failed: Funding ScriptSignatureParams did not match offer funding inputs
        at org.bitcoins.cli.ConsoleCli$.error$1(ConsoleCli.scala:1380)
        at org.bitcoins.cli.ConsoleCli$.$anonfun$exec$1(ConsoleCli.scala:1700)
        at scala.util.Try$.apply(Try.scala:210)
        at org.bitcoins.cli.ConsoleCli$.exec(ConsoleCli.scala:1642)
        at org.bitcoins.gui.dlc.DLCPaneModel.$anonfun$printDLCDialogResult$1(DLCPaneModel.scala:77)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at org.bitcoins.gui.TaskRunner$$anon$1.call(TaskRunner.scala:41)
        at javafx.concurrent.Task$TaskCallable.call(Task.java:1425)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.lang.Thread.run(Thread.java:829)
2021-05-04T18:40:27UTC INFO [ActorSystemImpl] http-rpc-server: Response for
```

**Comment by Christewart** [comment]
Fixed by #3040
